.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,241 @@
+          CIBW_SKIP: cp35-* pp*
+          CIBW_BEFORE_ALL_LINUX: cd snap7-full-1.4.2/build/unix/;make -f ${{ matrix.makefile }} install
+
+      - name: move shared library to ./lib
+        run: |
+          cd wheelhouse
+          for wheelname in $(ls *.whl)
+          do
+          dirname=$wheelname-dir
+          packagename=$(python3 -m wheel unpack $wheelname -d $dirname | sed -e 's/.*\/\(.*\)...OK/\1/')
+          mkdir -p $dirname/$packagename/snap7/lib
+          mv $dirname/$packagename/python_snap7.libs/libsnap7-*.so $dirname/$packagename/snap7/lib/libsnap7.so
+          rm -rf $dirname/$packagename/python_snap7.libs
+          rm $dirname/$packagename/snap7/__dummy__*.so
+          python3 -m wheel pack $dirname/$packagename -d ./
+          rm -rf $dirname
+          done
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: wheelhouse/*.whl
+
+  windows-build:
+    name: Build binary wheels for windows
+    runs-on: windows-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - arch: win_amd64
+            dll_dir: Win64
+          - arch: win32
+            dll_dir: Win32
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install choco packages
+        run: choco install 7zip wget python
+
+      - name: Cache snap7-archive
+        id: snap7-archive
+        uses: actions/cache@v2
+        with:
+          path: snap7-full-1.4.2.7z
+          key: ${{ env.snap7-archive-url }}
+
+      - name: Get snap7
+        if: steps.snap7-archive.outputs.cache-hit != 'true'
+        run: |
+          wget -O snap7-full-1.4.2.7z --content-disposition -c ${{ env.snap7-archive-url }}
+
+      - name: Extract archive
+        run: 7z x snap7-full-1.4.2.7z
+
+      - name: update wheel
+        run: python3 -m pip install wheel --upgrade
+
+      - name: build wheel
+        run: python3 setup.py bdist_wheel --plat-name ${{ matrix.arch }}
+
+      - name: move dll to ./lib
+        run: |
+          New-Item -ItemType Directory wheelhouse
+          Set-Location dist
+          $wheels = Get-ChildItem *.whl
+          foreach ($wheelname in $wheels) {
+            $dirname = -join($wheelname, "-dir")
+            $unpack_string = $(python3 -m wheel unpack $wheelname -d $dirname)
+            $unpack_string -match ".*: (?<pkgname>.*)...Ok"
+            $packagename = $matches['pkgname']
+            $libdir = -join($packagename,"\snap7\lib")
+            New-Item -ItemType Directory -Path $libdir
+            Copy-Item ..\snap7-full-1.4.2\release\Windows\${{ matrix.dll_dir }}\snap7.dll $libdir
+            python3 -m wheel pack $packagename -d ..\wheelhouse
+            Remove-Item $dirname -Force -Recurse
+          }
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: wheelhouse/*.whl
+
+  osx-build:
+    name: Build binary wheels for osx
+    runs-on: macos-10.15
```

Comment:
we might want to use pyenv to install multiple pythons

https://github.com/MonetDBSolutions/MonetDBe-Python/blob/master/.github/workflows/osx.yml#L25

Reply:
cibuildwheel action creates a docker container for each python version. So we don't have to.